validator/sawtooth_validator/journal/publisher.py

```diff
@@ -221,7 +222,7 @@ def on_check_publish_block(self, force=False):
                                                             _candidate_block):
                     candidate = self._candidate_block
                     self._candidate_block = None
-                    candidate = self._finalize_block(candidate)
+                    self._finalize_block(candidate)
```

Comment:
Is it find to call `builder.bulid_block()` at this point?  I changed this code because it wasn't following the doc from the consensus interfaces.